core/src/test/java/org/mule/el/mvel/datatype/AbstractVarAssignmentDataTypePropagatorTestCase.java

```diff
@@ -53,16 +57,42 @@ protected void doAssignmentDataTypePropagationTest(PropertyScope scope, String e
 
         MuleEvent testEvent = getTestEvent(TEST_MESSAGE);
 
+        CompiledExpression compiledExpression = compileMelExpression(expression, testEvent);
+
+        dataTypePropagator.propagate(testEvent.getMessage(), new TypedValue(TEST_MESSAGE, expectedDataType), compiledExpression);
+
+        assertThat(testEvent.getMessage().getPropertyDataType(PROPERTY_NAME, scope), like(String.class, JSON, CUSTOM_ENCODING));
+    }
+
+    protected void doInnerAssignmentDataTypePropagationTest(PropertyScope scope, String expression) throws Exception
+    {
+        final DataType expectedDataType = DataTypeFactory.create(Map.class, UNKNOWN);
+        expectedDataType.setEncoding(CUSTOM_ENCODING);
+
+        MuleEvent testEvent = getTestEvent(TEST_MESSAGE);
+        final Map<String, String> propertyValue = new HashMap<>();
+        propertyValue.put(INNER_PROPERTY_NAME, TEST_MESSAGE);
+        testEvent.getMessage().setProperty(PROPERTY_NAME, propertyValue, scope, expectedDataType);
+
+        CompiledExpression compiledExpression = compileMelExpression(expression, testEvent);
+
+        // Attempts to propagate a different dataType, which should be ignored
+        dataTypePropagator.propagate(testEvent.getMessage(), new TypedValue(propertyValue, DataType.STRING_DATA_TYPE), compiledExpression);
+
+        assertThat(testEvent.getMessage().getPropertyDataType(PROPERTY_NAME, scope), like(Map.class, UNKNOWN, CUSTOM_ENCODING));
+    }
+
+    private CompiledExpression compileMelExpression(String expression, MuleEvent testEvent)
+    {
         final ParserConfiguration parserConfiguration = MVELExpressionLanguage.createParserConfiguration(Collections.EMPTY_MAP);
         final MVELExpressionLanguageContext context = createMvelExpressionLanguageContext(testEvent, parserConfiguration);
 
         CompiledExpression compiledExpression = (CompiledExpression) compileExpression(expression, new ParserContext(parserConfiguration));
+
```

Comment:
All empty lines are unnecessary from the code point of view, is really interesting to me to know what makes this line "unnecesary" for you. For me, this line separates parts of the test setup code to make it more readeable (for me at least).